assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -832,6 +839,28 @@ public Assignment addAssignment(String context) throws PermissionException {
         return assignment;
     }
 
+    @Override
+    @Transactional
+    public void addAssignmentTimeSheet(AssignmentTimeSheet timeSheet, String context) throws PermissionException {
+        // security check
+        if (!allowAddSubmission(context)) {
+            throw new PermissionException(sessionManager.getCurrentSessionUserId(), SECURE_ADD_TIMESHEET, null);
+        }
+        assignmentRepository.newAssignmentTimeSheet(timeSheet);
+        log.debug("Created new timeSheet {}", timeSheet.getId());
+    }
+
+    @Override
+    @Transactional
+    public void removeAssignmentTimeSheet(AssignmentTimeSheet timeSheet, String context) throws PermissionException {
```

Comment:
You can load the timesheet and then get the siteId/context and then do the permission check.

Reply:
Done.
Removed the parameter.  Now the context is obtained from the timesheet.
`if (!allowAddSubmission(timeSheet.getSubmitter().getSubmission().getAssignment().getContext()))`